java/vector/src/test/java/org/apache/arrow/vector/TestVectorSchemaRoot.java

```diff
@@ -153,6 +154,81 @@ private VectorSchemaRoot createBatch() {
     return schemaRoot;
   }
 
+  @Test
+  public void testAddVector() {
+    try (final IntVector intVector1 = new IntVector("intVector1", allocator);
+         final IntVector intVector2 = new IntVector("intVector2", allocator);
+         final IntVector intVector3 = new IntVector("intVector3", allocator);) {
+
+      VectorSchemaRoot original = new VectorSchemaRoot(Arrays.asList(intVector1, intVector2));
+      assertEquals(2, original.getFieldVectors().size());
+
+      VectorSchemaRoot newRecordBatch = original.addVector(1, intVector3);
+      assertEquals(3, newRecordBatch.getFieldVectors().size());
+      assertEquals(intVector3, newRecordBatch.getFieldVectors().get(1));
+
+      original.close();
+      newRecordBatch.close();
+    }
+  }
+
+  @Test
+  public void testRemoveVector() {
+    try (final IntVector intVector1 = new IntVector("intVector1", allocator);
+        final IntVector intVector2 = new IntVector("intVector2", allocator);
+        final IntVector intVector3 = new IntVector("intVector3", allocator);) {
+
+      VectorSchemaRoot original =
+          new VectorSchemaRoot(Arrays.asList(intVector1, intVector2, intVector3));
+      assertEquals(3, original.getFieldVectors().size());
+
+      VectorSchemaRoot newRecordBatch = original.removeVector(0);
+      assertEquals(2, newRecordBatch.getFieldVectors().size());
+      assertEquals(intVector2, newRecordBatch.getFieldVectors().get(0));
+      assertEquals(intVector3, newRecordBatch.getFieldVectors().get(1));
+
+      original.close();
+      newRecordBatch.close();
+    }
+  }
+
+  @Test
+  public void testSlice() {
+    try (final IntVector intVector = new IntVector("intVector", allocator);
+         final Float4Vector float4Vector = new Float4Vector("float4Vector", allocator)) {
+      intVector.setValueCount(10);
+      float4Vector.setValueCount(10);
+      for (int i = 0; i < 10; i++) {
+        intVector.setSafe(i, i);
+        float4Vector.setSafe(i, i + 0.1f);
+      }
+      final VectorSchemaRoot original = new VectorSchemaRoot(Arrays.asList(intVector, float4Vector));
+
+      VectorSchemaRoot slice1 = original.slice(0, original.getRowCount());
+      assertEquals(original, slice1);
+
+      VectorSchemaRoot slice2 = original.slice(0, 5);
+      assertEquals(5, slice2.getRowCount());
+      // validate data
+      IntVector childVector1 = (IntVector) slice2.getFieldVectors().get(0);
+      Float4Vector childVector2 = (Float4Vector) slice2.getFieldVectors().get(1);
+      for (int i = 0; i < 5; i++) {
+        assertEquals(i, childVector1.get(i));
+        assertEquals(i + 0.1f, childVector2.get(i), 0);
+      }
+
+      // slice with invalid param
+      try {
```

Comment:
probably better to write another test with an ExpectedException (in case code does not throw the test will not catch it)

Reply:
Thanks a lot, fixe now, wait build turn green.